### PR TITLE
[BugFix] Fix broker load parallel performance issue since chunk buffer limit to one

### LINKED ISSUE #14695: BrokerLoad parallel load performance degradation
### Steps to reproduce the behavior (Required)
1. CREATE TABLE
2. ADMIN SET FRONTEND CONFIG("load_parallel_instance_num"="16")
3. BROKER LOAD

### Expected behavior (Required)
load using using more than 16 cpu cores

### Real behavior (Required)
load using less than 2 cpu cores

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 main

### PR DESCRIPTION
**Pull request by meegoo**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #14695

## Problem Summary(Required) ：

`query_mem_limit` default value is `0`, so that chunk_buffer_capacity always limit to 1. This will lead to performance degradation of broker load.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by meegoo** [comment]
run starrocks_be_unittest

File: be/src/exec/vectorized/connector_scan_node.cpp [hunk 1]
```diff
@@ -141,11 +141,11 @@ pipeline::OpFactories ConnectorScanNode::decompose_to_pipeline(pipeline::Pipelin
     size_t dop = context->dop_of_source_operator(id());
     std::shared_ptr<pipeline::ConnectorScanOperatorFactory> scan_op = nullptr;
 
-    if (config::connector_dynamic_chunk_buffer_limiter_enable) {
+    int64_t mem_limit = runtime_state()->query_mem_tracker_ptr()->limit() * config::scan_use_query_mem_ratio;
```
Comment (chaoyli) [inline be/src/exec/vectorized/connector_scan_node.cpp:144]: Add a issue and also add concise message in the commit message.

**Comment by meegoo** [comment]
run starrocks_be_unittest

**Review by Astralidea — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**